apps/degen-web-ui/pages/api/poap/event/start.ts

```diff
@@ -0,0 +1,49 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getSession } from 'next-auth/react';
+import { getDiscordService } from '../../../../src/core/api/discord.service';
+import { getPoapService } from '../../../../src/core/api/poap.service';
+import { DiscordSession } from '../../../../src/core/interfaces/auth-session';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const poapService = await getPoapService(req);
+  const discordService = await getDiscordService(req);
+  const userSession = (await getSession({ req })) as DiscordSession;
+
+  const { eventName, duration, guildId, voiceChannelId } = req.body;
+  try {
+    const isPoapAdmin = await poapService.isPoapAdmin(guildId, userSession?.user?.id);
+
+    if (!isPoapAdmin) {
+      res.status(401).json({
+        message: `Must Be Poap Admin to start events`,
+      });
+      return;
+    }
+
+    const event = await poapService.startPoapEvent(
+      eventName,
+      duration,
+      userSession.user?.id,
+      guildId,
+      voiceChannelId
+    );
+
+    // Get list of current participants
+    const members = await discordService.getMembersInChannel(guildId, voiceChannelId);
+
+    await poapService.removePoapParticipants(guildId, voiceChannelId);
+
+    members.forEach((member) => {
+      poapService.insertPoapParticipant(member, eventName);
+    });
```

Comment:
need to do this with a map in a Promise.all to handle the async results correctly instead like this. as is this will finish after the result is returned.